### PR TITLE
[#176013578] Remove prepare script

### PR DESCRIPTION
**Pull request by balanza**
`prepare` script is executed before and every time we run `npm publish`. We used to build the package in the `prepare` script, and this makes perfect sense when publishing from local environment.

As we moved the publication on the CI/CD environment, and we separated build and publish step, that is no longer needed.
As double check, I introduced a `prepack` script which basically confirms that we are shipping every needed file in the package.

The script is in bash: that introduces the constraint that we can only publish from environment that support such language. My assumption is that it's a reasonable restriction.

Plus: i removed `shx` as we no longer need it.

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -98,24 +98,13 @@ stages:
             inputs:
               SourceFolder: '$(System.DefaultWorkingDirectory)'
               TargetFolder: '$(System.DefaultWorkingDirectory)/bundle'
+              # The list of files to be considered is determined by the "files" entry in package.json
```
Comment (gquadrati) [inline .devops/deploy-pipelines.yml:101]: what do you mean?

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "dist/"
   ],
   "scripts": {
-    "prepare": "yarn build",
+    "prepack": "[[ ! -d ./dist ]] && echo \"Cannot create package: missing 'dist' folder\" && exit 1",
```
Comment (gquadrati) [inline package.json:12]: where this "prepack" script will be called?

File: .devops/deploy-pipelines.yml (hunk 1 above)
Comment (gquadrati) [inline .devops/deploy-pipelines.yml:101]: what do you mean?

**Review by gquadrati — APPROVED** [review]
LGTM

File: package.json (hunk 2 above)
Comment (balanza) [reply package.json:12]: Before `npm pack` is executed, and that includes `npm publish`. See https://docs.npmjs.com/cli/v6/using-npm/scripts#prepare-and-prepublish

File: .devops/deploy-pipelines.yml (hunk 1 above)
Comment (balanza) [reply .devops/deploy-pipelines.yml:101]: this: https://github.com/pagopa/io-functions-commons/blob/ca9a943a9004bbf7aec7804c5715fbff80ab8fde/package.json#L8

Comment (gquadrati) [reply .devops/deploy-pipelines.yml:101]: yes, I saw it after this comment.. sorry about that!

File: package.json (hunk 2 above)
Comment (gquadrati) [reply package.json:12]: thanks!

**Pull request merged**